Source/WebCore/platform/mediastream/gstreamer/GStreamerMockDevice.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2023 Metrological Group B.V.
+ * Author: Philippe Normand <philn@igalia.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * aint with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+
+#if ENABLE(MEDIA_STREAM) && USE(GSTREAMER)
+#include "GStreamerMockDevice.h"
+
+#include "CaptureDevice.h"
+#include "GRefPtrGStreamer.h"
+#include "GStreamerCommon.h"
+#include "GUniquePtrGStreamer.h"
+#include <wtf/glib/WTFGType.h>
+
+using namespace WebCore;
+
+struct _GStreamerMockDevicePrivate {
+};
+
+GST_DEBUG_CATEGORY_STATIC(webkitGstMockDeviceDebug);
+#define GST_CAT_DEFAULT webkitGstMockDeviceDebug
```

Comment:
Where is this initialized?

Reply:
In the `WEBKIT_DEFINE_TYPE_WITH_CODE` macro expansion.